cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -25,4 +33,49 @@ export class AwsDefaults {
         }
         return nodeTypeList[Math.round(nodeTypeList.length / 2)];
     };
+
+    static defaulAvailabilityZoneNameStrategy = (azList: { [key: string]: string }[], nodeProfile: string) => {
+        const defaultAZNameList: { [key: string]: string }[] = [];
+        switch (nodeProfile) {
+            case AWS_NODE_PROFILE_NAMES.SINGLE_NODE: {
+                return [azList[0]];
+            }
+            default: {
+                if (azList.length >= 3) {
+                    azList.slice(0, 3).forEach((az) => {
+                        defaultAZNameList.push(az);
+                    });
+                } else {
+                    console.error('There is not enough Availability Zone');
```

Comment:
I suggest including the nodeProfile in this message, something like:
`For profile ${nodeProfile}, we expect to use 3 node profiles, but azList has a length of ${azList.length}`

Reply:
Done